pkg/reaper/resource.go

```diff
@@ -0,0 +1,147 @@
+	return kluster.Name + "-" + dc.Name + "-reaper"
+}
+
+func NewReaper(reaperKey types.NamespacedName, kc *k8ssandraapi.K8ssandraCluster, reaperTemplate *api.ReaperTemplate, actualDc *cassdcapi.CassandraDatacenter) *api.Reaper {
+	labels := createResourceLabels(kc)
+	desiredReaper := &reaperapi.Reaper{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   reaperKey.Namespace,
+			Name:        reaperKey.Name,
+			Annotations: map[string]string{},
+			Labels:      labels,
+		},
+		Spec: reaperapi.ReaperSpec{
+			ReaperTemplate: *reaperTemplate,
+			DatacenterRef: reaperapi.CassandraDatacenterRef{
+				Name:      actualDc.Name,
+				Namespace: actualDc.Namespace,
+			},
+		},
+	}
+	return desiredReaper
+}
+
+func SetDefaults(reaper *api.Reaper) bool {
+	updated := false
+	if reaper.Spec.Image == "" {
+		reaper.Spec.Image = DefaultReaperImage
+		updated = true
+	}
+	if reaper.Spec.ImagePullPolicy == "" {
+		reaper.Spec.ImagePullPolicy = DefaultImagePullPolicy
+		updated = true
+	}
+	if reaper.Spec.Keyspace == "" {
+		reaper.Spec.Keyspace = DefaultKeyspace
+		updated = true
+	}
+	if reaper.Spec.ServiceAccountName == "" {
+		reaper.Spec.ServiceAccountName = "default"
+		updated = true
+	}
+	return updated
+}
+
+// Coalesce combines the cluster and dc templates with override semantics. If a property is
+// defined in both templates, the dc-level property takes precedence.
+func Coalesce(clusterTemplate *api.ReaperTemplate, dcTemplate *api.ReaperTemplate) *api.ReaperTemplate {
+
+	if clusterTemplate == nil && dcTemplate == nil {
+		return nil
+	}
+
+	coalesced := &api.ReaperTemplate{}
+
+	if dcTemplate != nil && len(dcTemplate.Image) != 0 {
+		coalesced.Image = dcTemplate.Image
+	} else if clusterTemplate != nil && len(clusterTemplate.Image) != 0 {
+		coalesced.Image = clusterTemplate.Image
+	}
+
+	if dcTemplate != nil && len(dcTemplate.ImagePullPolicy) != 0 {
+		coalesced.ImagePullPolicy = dcTemplate.ImagePullPolicy
+	} else if clusterTemplate != nil && len(clusterTemplate.ImagePullPolicy) != 0 {
+		coalesced.ImagePullPolicy = clusterTemplate.ImagePullPolicy
+	}
+
+	if dcTemplate != nil && len(dcTemplate.ServiceAccountName) != 0 {
+		coalesced.ServiceAccountName = dcTemplate.ServiceAccountName
+	} else if clusterTemplate != nil && len(clusterTemplate.ServiceAccountName) != 0 {
+		coalesced.ServiceAccountName = clusterTemplate.ServiceAccountName
+	}
+
+	if dcTemplate != nil && len(dcTemplate.Keyspace) != 0 {
+		coalesced.Keyspace = dcTemplate.Keyspace
+	} else if clusterTemplate != nil && len(clusterTemplate.Keyspace) != 0 {
+		coalesced.Keyspace = clusterTemplate.Keyspace
+	}
+
+	if dcTemplate != nil && len(dcTemplate.CassandraUserSecretRef) != 0 {
+		coalesced.CassandraUserSecretRef = dcTemplate.CassandraUserSecretRef
+	} else if clusterTemplate != nil && len(clusterTemplate.CassandraUserSecretRef) != 0 {
+		coalesced.CassandraUserSecretRef = clusterTemplate.CassandraUserSecretRef
+	}
+
+	if dcTemplate != nil && len(dcTemplate.JmxUserSecretRef) != 0 {
```

Comment:
If `JmxUserSecretRef` is `nil`, then K8ssandraClusterReconciler should create the user similar to how/where it creates the default superuser. It should get copied to remote clusters via the ReplicatedSecret.

If `JmxSecretRef` is not `nil` then the necessary labels (see [here](https://github.com/k8ssandra/k8ssandra-operator/blob/main/pkg/secret/replicated.go#L133)) should be added to it so that it gets replicated.

Also note that remote JMX needs to be configured. See [here](https://github.com/k8ssandra/k8ssandra/blob/main/charts/k8ssandra/templates/cassandra/cassdc.yaml#L226) for how it is done in the k8ssandra Helm chart.